HpToolsLauncher/Launcher.cs

```diff
@@ -886,24 +886,30 @@ private void RunTests(IAssetRunner runner, string resultsFile, TestSuiteRunResul
                 }
 
                 //if there is an error
-                if (results.TestRuns.Any(tr => tr.TestState == TestState.Failed || tr.TestState == TestState.Error))
+                if (results.TestRuns.Any(tr => tr.TestState == TestState.Error))
                 {
                     _exitCode = ExitCodeEnum.Failed;
                 }
 
                 int numFailures = results.NumFailures;
                 int numSuccess = results.TestRuns.Count(t => t.TestState == TestState.Passed);
                 int numErrors = results.NumErrors;
-                int numWarnings = results.TestRuns.Count(t => t.TestState == TestState.Warning);
+                int numWarnings = results.NumWarnings;
 
-                if ((numErrors <= 0) && (numFailures > 0) && _exitCode != ExitCodeEnum.Failed)
-                {
-                    _exitCode = ExitCodeEnum.Failed;
-                }
-
-                if ((numErrors <= 0) && (numFailures > 0) && (numSuccess > 0) && _exitCode != ExitCodeEnum.Failed)
-                {
-                    _exitCode = ExitCodeEnum.Unstable;
+                if (_exitCode != ExitCodeEnum.Failed && _exitCode != ExitCodeEnum.Aborted)
+				{
+                    if ((numErrors <= 0) && (numFailures > 0) && (numSuccess > 0))
+                    {
+                        _exitCode = ExitCodeEnum.Unstable;
+                    }
```

Comment:
If an error occurred through the job and its status is already set before to FAILED, it should remain FAILED and not overwritten. The same is valid for ABORTED state as well.

Reply:
It makes sense to check the exit code `Aborted` since once set it shall not be changed any more, however, for the `Failed` code, it shall not be checked here.

The `RunTests` function will call in two places. The first one is the normal run, which reflects the test run results. The second place is **rerun** and the `Failed` code may become `Unstable`. For example, if a test is failed first time and after rerun it is passed, then the final code shall be `Unstable` rather than `Failed. You changes may always set `Failed` in this scenario.

Reply:
But how should we differentiate a situation when because a non-existing test set an error has occurred? I set the statement as is, to eliminate the possibility of overwriting such situation, but I completely agree with you on this point. If I modify the code, if an error has occurred of non-existing test set(s), it can be overwritten to `Unstable`.

Reply:
_Update_: only the `Aborted` state will not be overwritten.

Reply:
Looks good